virtual_rainforest/soil/model.py

```diff
@@ -40,57 +36,67 @@ class SoilModel(BaseModel, model_name="soil"):
     def __init__(self, update_interval: timedelta64, no_layers: int, **kwargs: Any):
 
         if no_layers < 1:
-            LOGGER.error("There has to be at least one soil layer in the soil model!")
+            log_and_raise(
+                "There has to be at least one soil layer in the soil model!",
+                InitialisationError,
+            )
         elif no_layers != int(no_layers):
-            LOGGER.error("The number of soil layers must be an integer!")
+            log_and_raise(
+                "The number of soil layers must be an integer!", InitialisationError
+            )
 
         super(SoilModel, self).__init__(update_interval, **kwargs)
         self.no_layers = int(no_layers)
         # Save variables names to be used by the __repr__
         self._repr.append("no_layers")
 
     @classmethod
-    def factory(cls, config: dict[str, Any]) -> SoilModel:
+    def factory(cls, config: dict[str, Any]) -> Union[SoilModel, FailedModel]:
```

Comment:
A factory method should always return the initialised model or fail with an exception. Returning something different is a code smell and likely to result in issues. It is not needed, as I suggest above.